### PR TITLE
URL Connector: Adding "webcrawler" ConnectorProvider

### PR DESCRIPTION
**Pull request by lasryaric**
Adding the `webcrawler` in `types/..../ConnectorProvider` and all the data structured defining it.

### CODE REVIEW

File: front/lib/data_sources.ts [hunk 1]
```diff
@@ -1,36 +1,4 @@
-import { CoreAPIDocument, DataSourceType } from "@dust-tt/types";
-
-export function getProviderLogoPathForDataSource(
```
Comment (lasryaric) [inline front/lib/data_sources.ts:3]: Reason for deleting this code:
- The search page is not used anymore.
- We are not supposed to use these logo anymore, we are supposed to use Icons instead

File: connectors/src/connectors/index.ts [hunk 2]
```diff
@@ -77,15 +78,26 @@ import { launchSlackSyncWorkflow } from "@connectors/connectors/slack/temporal/c
...
 import logger from "@connectors/logger/logger";
 
+import {
+  cleanupWebcrawlerConnector,
+  createWebcrawlerConnector,
+  retrieveWebcrawlerConnectorPermissions,
+  retrieveWebCrawlerObjectsParents,
+  retrieveWebCrawlerObjectsTitles,
+  stopWebcrawlerConnector,
+} from "./webcrawler";
```
Comment (fontanierh) [inline connectors/src/connectors/index.ts:88]: nit: relative imports

File: connectors/src/connectors/webcrawler/index.ts [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+import {
+  WebCrawlerConfiguration,
+  WebCrawlerFolder,
+  WebCrawlerPage,
+} from "@connectors/lib/models/webcrawler";
+import { Err, Ok, type Result } from "@connectors/lib/result.js";
+import logger from "@connectors/logger/logger";
+import type { DataSourceConfig } from "@connectors/types/data_source_config.js";
+
+import { ConnectorPermissionRetriever } from "../interface";
```
Comment (fontanierh) [inline connectors/src/connectors/webcrawler/index.ts:14]: nit

**Review by fontanierh — APPROVED** [review]
LGTM

File: front/pages/w/[wId]/builder/data-sources/[name]/index.tsx [hunk 4]
```diff
@@ -926,10 +944,27 @@ function ManagedDataSourceView({
...
+                      return (
+                        <>
+                          Selected resources will be accessible to all members
+                          of the workspace.
+                          <br />
+                          Changes may impact existing assistants.
+                        </>
+                      );
+                    case "webcrawler":
+                      return null;
```
Comment (spolu) [inline front/pages/w/[wId]/builder/data-sources/[name]/index.tsx:963]: why null here?

**Pull request merged**